cli/internal/cloudcmd/iam_test.go

```diff
@@ -148,3 +153,198 @@ func TestIAMCreator(t *testing.T) {
 		})
 	}
 }
+
+func TestDestroyIAMConfiguration(t *testing.T) {
+	newError := func() error {
+		return errors.New("failed")
+	}
+
+	testCases := map[string]struct {
+		tfClient                   terraformClient
+		wantErr                    bool
+		wantDestroyCalled          bool
+		wantCleanupWorkspaceCalled bool
+	}{
+		"destroy error": {
+			tfClient:          &stubTerraformClient{destroyErr: newError()},
+			wantErr:           true,
+			wantDestroyCalled: true,
+		},
+		"destroy": {
+			tfClient:                   &stubTerraformClient{},
+			wantDestroyCalled:          true,
+			wantCleanupWorkspaceCalled: true,
+		},
+		"cleanup error": {
+			tfClient:                   &stubTerraformClient{cleanUpWorkspaceErr: newError()},
+			wantErr:                    true,
+			wantDestroyCalled:          true,
+			wantCleanupWorkspaceCalled: true,
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			assert := assert.New(t)
+			destroyer := &IAMDestroyer{client: tc.tfClient}
+
+			err := destroyer.DestroyIAMConfiguration(context.Background())
+
+			if tc.wantErr {
+				assert.Error(err)
+			} else {
+				assert.NoError(err)
+			}
+			if tc.wantDestroyCalled {
+				assert.True(destroyer.client.(*stubTerraformClient).destroyCalled)
+			} else {
+				assert.False(destroyer.client.(*stubTerraformClient).destroyCalled)
+			}
+			if tc.wantCleanupWorkspaceCalled {
+				assert.True(destroyer.client.(*stubTerraformClient).cleanUpWorkspaceCalled)
+			} else {
+				assert.False(destroyer.client.(*stubTerraformClient).cleanUpWorkspaceCalled)
+			}
```

Comment:
```suggestion
			if tc.wantErr {
				assert.Error(err)
			} else {
				assert.NoError(err)
			}
			assert.Equal(tc.wantDestroyCalled, tc.tfClient.destroyCalled)
			assert.Equal(tc.wantCleanupWorkspaceCalled, tc.tfClient.cleanUpWorkspaceCalled)
```
You shouldn't just cast things, as this can panic at runtime. I'd suggest to just change the type of tc.tfClient to the concrete type.